### DISCUSSION TITLE
Yellowlist, surrogate, or widget?

### DISCUSSION THREAD

**Discussion by Farchanter** [EFForg/privacybadger#2740] [Q&A]
I found a [page](https://today.tcnj.edu/) that doesn't work with Privacy Badger (or, at least, not Privacy Badger as it has evolved for me) because of some scripts it loads in from a third-party called Tagboard. Playing around changing some of those scripts from red to yellow but leaving others alone, I think I was able to get the intended content on the page to load without tracking.

I'm interested in trying to fix this for the larger project, but I'm uncertain about what my first steps should be. Is it possible to yellowlist some scripts from a domain while leaving others blocked, does it make sense to load in only the helpful scripts as a "surrogate", or should I work on a broader replacement widget for Tagboard?

Thanks for any guidance!

**Comment by ghostwords** [comment]
Hello! It sounds like you've already seen our [guide to fixing broken sites](https://github.com/EFForg/privacybadger/blob/master/doc/fixing-broken-sites.md), but it wasn't clear which path you should take.

Tagboard looks like a provider of social media feed widgets. We could add some/all of `tagboard.com` domains to the "yellowlist", and as you've confirmed by manually toggling the sliders to yellow, that should resolve the breakages.

However, we give up some privacy by still letting `tagboard.com` domains load. Instead of yellowlisting, we could try to add a click-to-activate replacement for Tagboard instead. Should we though? Let's take Disqus (comments widget provider) as an example. Disqus widgets are a good candidate for replacement since
- Disqus shows up on many websites across the Web (they track you by default across all those sites without consent)
- The functionality Disqus provides isn't essential to the page (you want to read the news story but not necessarily the comments underneath)
- At the same time you may in fact want to read the comments, at least sometimes
- The widgets have a mostly predictable way they work (Disqus uses the same script URL and the same DOM element name/ID on most websites it shows up on)

If the pages the widget shows up on mostly require the widget (you would always want the widget active since there is nothing else there otherwise), it's probably not a good fit for our widget replacement system. It's also probably not worth our time to provide replacements for obscure widgets and/or widgets that are hard to replace for whatever reason.

"Surrogate" scripts are meant to be a seamless replacement for JavaScript trackers like Google Analytics that provide no useful functionality to the user but when blocked sometimes cause websites to break (for example when the website's navigation links require the tracker's JavaScript to be present in order to function). To work around such breakages, we supply blank versions of tracking scripts that look like real tracking scripts to the website but perform no actual tracking.

Let me know if this helped.
